src/MSBuild/TerminalLogger/NodeStatus.cs

```diff
@@ -13,13 +14,47 @@ internal class NodeStatus
 {
     public string Project { get; }
     public string? TargetFramework { get; }
+    public TerminalColor TargetPrefixColor { get; } = TerminalColor.Default;
+    public string? TargetPrefix { get; }
     public string Target { get; }
     public StopwatchAbstraction Stopwatch { get; }
 
+    /// <summary>
+    /// Status of a node that is currently doing work.
+    /// </summary>
+    /// <param name="project">The project that is written on left side.</param>
+    /// <param name="targetFramework">Target framework that is colorized and written on left side after project.</param>
+    /// <param name="target">The currently running work, usually the currently running target. Written on right.</param>
+    /// <param name="stopwatch">Duration of the current step. Written on right after target.</param>
     public NodeStatus(string project, string? targetFramework, string target, StopwatchAbstraction stopwatch)
+    {
+#if DEBUG
```

Comment:
NIT: Please consider to use `Debug.Assert` instead. I believe it is more consistent with our code base

Reply:
I did in the past, but it will show a pop-up dialog in tests and block test execution when running in VS. And the test result depends on which button you click in the Assert UI.